pkgs/stdenv/cross/default.nix

```diff
@@ -1,14 +1,14 @@
-{ system, allPackages, platform, crossSystem, config, ... } @ args:
+{ lib, allPackages
+, system, platform, crossSystem, config
+}:
 
 rec {
-  argClobber = {
+  vanillaStdenv = (import ../. {
+    inherit lib allPackages system platform;
     crossSystem = null;
     # Ignore custom stdenvs when cross compiling for compatability
     config = builtins.removeAttrs config [ "replaceStdenv" ];
-  };
-  vanillaStdenv = (import ../. (args // argClobber // {
-    allPackages = args: allPackages (argClobber // args);
```

Comment:
Note:
This is fine, because, as the stdenv are now explicitly forced to provide a `config` attribute, by the fact that this commit removes the `config` value from `top-level/default.nix`. This `config` attribute would necessary be the `config` attribute given as argument to the `vanillaStdenv`.  So there is no need to make a custom version of `allPackages` here.